### PR TITLE
add some nginx WAF documentation

### PR DESCRIPTION
**Pull request by fabianfrz**

### CODE REVIEW

File: source/manual/how-tos/nginx_waf.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
+===============================
+nginx: Web Application Firewall
+===============================
+
+.. Warning::
+
+    This page is for advanced users only. A misconfiguration here may break your page.
```
Comment (fichtner) [inline source/manual/how-tos/nginx_waf.rst]: “Your page” needs clarification

File: source/manual/how-tos/nginx_waf.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Background Information
+======================
+
+The OPNsense WAF uses NAXSI_, which is a loadable module for the nginx_ web server.
+NAXSI has two rule types:
+
+.. _NAXSI: https://github.com/nbs-system/naxsi
+.. _nginx: https://nginx.org/
+
+* Main Rules: This rules are globally valid. Usual use case: Blocking stuff.
```
Comment (fichtner) [inline source/manual/how-tos/nginx_waf.rst]: “Stuff” needs clarification for both points

File: source/manual/how-tos/nginx_waf.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+* Basic Rules: This rules are usually used in the locations to whitelist stuff inside a location or for additional rules.
+
+In addition to the self defined rules, NAXSI contains libinjection which is available directly in the location configuration.
+The NAXSI project itself has a high quality documentation for the module online_.
+A good ruleset_ to start can be found at GitHub on the project page.
+It cannot be provided by the plugin because if license issues (GPL cannot be used in BSD 2 Clause code),
```
Comment (fichtner) [inline source/manual/how-tos/nginx_waf.rst]: Of

File: source/manual/how-tos/nginx_waf.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+=============
+
+WAF Rules
+---------
+
+.. image:: images/nginx_waf_rule.png
+
+WAF rules are used to trigger an action if a condition evaluates to true or false (negated).
+The usual use case is increasing a score which can be checked afterwards, but a rule can for example also block instantly (the plugin only supports a score).
+WAF rules are grouped to a WAF policy. which then can evaluate the aggregated score.
```
Comment (fichtner) [inline source/manual/how-tos/nginx_waf.rst]: Comma, not dot

File: source/manual/how-tos/nginx_waf.rst [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+Value    a compare value
+Operator choose a compare Operator to compare score operator value
+Action   usually block
+======== =========================================================
+
+As an (incomplete) example:
+
+======== ===============================================================
+Name     block SQLi
+Rules    contains select, contains from, contains union, containd delete
```
Comment (fichtner) [inline source/manual/how-tos/nginx_waf.rst]: Contains

**Comment by fabianfrz** [comment]
@fichtner I added your requested changes. For clarification I've added links for more detailed information and a warning on top of the page.

**Comment by fabianfrz** [comment]
yes the log viewer is here: opnsense/plugins#805

It is almost ready but needs some refactoring to clean it up (move model code from view to model, split log lines view etc.)

**Pull request merged**